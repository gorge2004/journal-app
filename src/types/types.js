export const types = {
    login: '[Auth] login',
    logout: '[Auth] logout',
    
    uiSetError: '[UI] SET ERROR',
    uiRemoveError: '[UI] Remove ERROR',
    uiStartLoading: '[UI] Start Loading',
    uiFinishLoading: '[UI] Finish Loading',
    
    notesAddNew: '[Notes] new note',
    notesActive: '[Notes] set active note',
    notesLoad: '[Notes] Load notes',
    notesUpdated: '[Notes] Update note',
    notesFileUrl: '[Notes] Update image url',
    notesDelete: '[Notes] Delete note',
    notesLogoutCleaning: '[Notes] logout Cleaning',
};