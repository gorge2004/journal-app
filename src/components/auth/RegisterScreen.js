import { Link } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";

import { useForm } from "../../hooks/useForm";
import validator from "validator";
import { removeError, setError } from "../../actions/ui";
import { startRegisterWithEmailAndPassword } from "../../actions/auth";
const RegisterScreen = () => {
  const [{ name, email, password, password2 }, handleInputchange, reset] =
    useForm({
      name: "Jorge",
      email: "jorge@gmail.com",
      password: "123456",
      password2: "123456",
    });
  // obetengo el state del redux tal como se tiene
  const { msgError } = useSelector((state) => state.ui);
  const dispatch = useDispatch();
  const handleSubmitRegister = (evt) => {
    evt.preventDefault();
    if (isFormValue()) {
      console.log(name, email, password, password2);
      dispatch(startRegisterWithEmailAndPassword(email, password, name));
      
    }
  };
  const isFormValue = () => {
    if (name.trim().length === 0) {
      dispatch(setError("name is required."));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(setError("email is not validate."));
      return false;
    } else if (password !== password2 || password2.length <= 5) {
      dispatch(
        setError("password should be at least characters and match each other.")
      );
      return false;
    }
    dispatch(removeError());

    return true;
  };
  return (
    <>
      <h3 className="auth__title">Register</h3>

      <form onSubmit={handleSubmitRegister} className="animate__animated animate__fadeIn animate__faster">
        {msgError && <div className="auth__alert-error">{msgError}</div>}

        <input
          className="auth__input"
          autoComplete="off"
          type="text"
          placeholder="Name"
          name="name"
          value={name}
          onChange={handleInputchange}
        />
        <input
          className="auth__input"
          autoComplete="off"
          type="text"
          placeholder="Email"
          name="email"
          value={email}
          onChange={handleInputchange}
        />
        <input
          className="auth__input"
          autoComplete="off"
          type="password"
          placeholder="Password"
          name="password"
          value={password}
          onChange={handleInputchange}
        />
        <input
          className="auth__input"
          autoComplete="off"
          type="password"
          placeholder="Confirm password"
          name="password2"
          value={password2}
          onChange={handleInputchange}
        />
        <button className=" btn btn-primary btn-block mb-5" type="submit">
          Login
        </button>

        <Link className="link mt-5" to="/auth/login">
          already registered ?
        </Link>
      </form>
    </>
  );
};

export default RegisterScreen;
