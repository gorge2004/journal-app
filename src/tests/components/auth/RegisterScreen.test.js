import { mount } from "enzyme";
import { Provider } from "react-redux";
import RegisterScreen from "../../../components/auth/RegisterScreen";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { MemoryRouter } from "react-router-dom";
import { types } from "../../../types/types";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {
  auth: {},
  ui: {
    msgError: null,
    loading: false,
  },
};
let store = mockStore(initialState);
const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <RegisterScreen />
    </MemoryRouter>
  </Provider>
);
describe("testing_RegisterScreen", () => {
  test("should_render_registerScreen_component", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should_call_to_registerAction", () => {
    const emailField = wrapper.find('input[name="email"]');
    emailField.simulate("change", {
      target: {
        value: "",
        name: "email",
      },
    });

    wrapper.find("form").simulate("submit", {
      preventDefault() {},
    });
    const actions = store.getActions();
    expect(actions[0].type).toBe(types.uiSetError);
  });

  test("should_render_error_box", () => {
    const initialState = {
      auth: {},
      ui: {
        msgError: 'Email no es correcto',
        loading: false,
      },
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <RegisterScreen />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.find('.auth__alert-error').exists()).toBe(true);
  });
});
