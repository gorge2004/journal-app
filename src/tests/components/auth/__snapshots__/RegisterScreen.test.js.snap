// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing_RegisterScreen should_render_registerScreen_component 1`] = `
Array [
  <h3
    className="auth__title"
  >
    Register
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Jorge"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="jorge@gmail.com"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123456"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value="123456"
    />
    <button
      className=" btn btn-primary btn-block mb-5"
      type="submit"
    >
      Login
    </button>
    <a
      className="link mt-5"
      href="/auth/login"
      onClick={[Function]}
    >
      already registered ?
    </a>
  </form>,
]
`;
